name-emu/src/main.rs

```diff
@@ -63,15 +69,14 @@ type DynResult<T> = std::result::Result<T, Box<dyn std::error::Error>>;
 //     }
 // }
 
-fn reset_mips() -> Mips {
+fn reset_mips(program_data: &[u8]) -> Mips {
   // Reset execution and begin again.
-  let mut mips: Mips = Default::default();
-
-  let program_data = std::fs::read("/home/qwe/Documents/CS4485/name/name-as/output.o").unwrap();
+  let mut mips: Mips = Default::default();  
 
   for (i, byte) in program_data.iter().enumerate() {
     mips.write_b(mips::DOT_TEXT_START_ADDRESS + i as u32, *byte).unwrap();
   }
+  mips.stop_address = mips::DOT_TEXT_START_ADDRESS as usize + program_data.len();
```

Comment:
Note for the future: we need some `name-const` module that we can read from so that we don't have multiples of stuff like the text start address

Reply:
yes, and that extends to lineinfo too, since currently I'm copy/pasting that struct verbatim. Call it name-common maybe?

Reply:
Agreed